game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -118,7 +118,7 @@ class ResourceType(grid: Grid,
 
 class Resource(var position: Vector2 = new Vector2(0f, 0f),
                val resourceMass: Float,
-               val scoreModification: Int = 0) extends Entity {
+               override val _scoreModification: Float = 0.0f) extends Entity {
```

Comment:
That's a tricky question. I'm not 100% sure I took the right decision. The thing is, `_scoreModification` is a `val` so I can't simply change its value, I need to do an override during the construction time. But, it's a parameter of the class with a default of `0f`. So any resource (normal one for example) can change this value when creating the object.